Bjs2 53972/добавить запрос на наставничество новый

MaxDrattcev opened the pull request:
Устранил замечания.

MaxDrattcev committed:
Запрос на менторство: JS2-53972

MaxDrattcev committed:
Запрос на менторство: JS2-53972. Устранил замечания.

MaxDrattcev committed:
Запрос на менторство: JS2-53972. Устранил замечания.

DmShevchuk commented on src/main/java/school/faang/user_service/controller/MentorshipRequestController.java:
```diff
@@ -0,0 +1,60 @@
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/mentorship")
```
@RequestMapping("/v1/mentorship")

DmShevchuk commented on src/main/java/school/faang/user_service/controller/MentorshipRequestController.java:
```diff
@@ -0,0 +1,60 @@
+public class MentorshipRequestController {
+    private final MentorshipRequestService mentorshipRequestService;
+
+    @PostMapping("/api/mentorship/requests")
```
Просто PostMapping

DmShevchuk commented on src/main/java/school/faang/user_service/controller/MentorshipRequestController.java:
```diff
@@ -0,0 +1,60 @@
+        return ResponseEntity.ok(response);
+    }
+
+    @GetMapping("/api/mentorship/filters")
```
Просто GetMapping

DmShevchuk commented on src/main/java/school/faang/user_service/controller/MentorshipRequestController.java:
```diff
@@ -0,0 +1,60 @@
+
+    @GetMapping("/api/mentorship/filters")
+    public ResponseEntity<List<MentorshipRequestDto>> getRequests(
+            @RequestParam(required = false) String description,
+            @RequestParam(required = false) Long requesterId,
+            @RequestParam(required = false) Long receiverId,
+            @RequestParam(required = false) RequestStatus status
```
Принимать в контроллера RequestFilterDto

DmShevchuk commented on src/main/java/school/faang/user_service/controller/MentorshipRequestController.java:
```diff
@@ -0,0 +1,60 @@
+            @RequestParam(required = false) RequestStatus status
+    ) {
+
+        RequestFilterDto filters = new RequestFilterDto();
+        filters.setDescription(description);
+        filters.setRequesterId(requesterId);
+        filters.setReceiverId(receiverId);
+        filters.setStatus(status);
```
Убрать

DmShevchuk commented on src/main/java/school/faang/user_service/controller/MentorshipRequestController.java:
```diff
@@ -0,0 +1,60 @@
+    @PostMapping("/api/mentorship/requests")
+    public ResponseEntity<MentorshipRequestDto> requestMentorship(@Valid @RequestBody MentorshipRequestDto mentorshipRequestDto) {
+        MentorshipRequestDto response = mentorshipRequestService.requestMentorship(mentorshipRequestDto);
+        return ResponseEntity.ok(response);
```
Просто MentorshipRequestDto, без ResponseEntity

DmShevchuk commented on src/main/java/school/faang/user_service/controller/MentorshipRequestController.java:
```diff
@@ -0,0 +1,60 @@
+
+        List<MentorshipRequestDto> requests = mentorshipRequestService.getRequests(filters);
+
+        return ResponseEntity.ok(requests);
```
Просто List<MentorshipRequestDto>, без ResponseEntity

DmShevchuk commented on src/main/java/school/faang/user_service/controller/MentorshipRequestController.java:
```diff
@@ -0,0 +1,60 @@
+        return ResponseEntity.ok(requests);
+    }
+
+    @PutMapping("/api/mentorship/{id}/accept")
```
@PutMapping("/{id}/acceptance")

DmShevchuk commented on src/main/java/school/faang/user_service/controller/MentorshipRequestController.java:
```diff
@@ -0,0 +1,60 @@
+    @PutMapping("/api/mentorship/{id}/accept")
+    public ResponseEntity<String> acceptRequest(@PathVariable long id) {
+        mentorshipRequestService.acceptRequest(id);
+        return ResponseEntity.ok("Запрос на менторство успешно принят.");
```
Вернуть DTO, без респонс ентити

DmShevchuk commented on src/main/java/school/faang/user_service/controller/MentorshipRequestController.java:
```diff
@@ -0,0 +1,60 @@
+        return ResponseEntity.ok("Запрос на менторство успешно принят.");
+    }
+
+    @PutMapping("/api/mentorship/{id}/reject")
```
@PutMapping("/{id}/rejection")

DmShevchuk commented on src/main/java/school/faang/user_service/controller/MentorshipRequestController.java:
```diff
@@ -0,0 +1,60 @@
+            @Valid @RequestBody RejectionDto rejection
+    ) {
+        mentorshipRequestService.rejectRequest(id, rejection);
+        return ResponseEntity.ok("Запрос на менторство успешно отклонен.");
```
Вернуть DTO, без респонс ентити

DmShevchuk commented on src/main/java/school/faang/user_service/service/MentorshipRequestService.java:
```diff
@@ -0,0 +1,108 @@
+    public MentorshipRequestDto requestMentorship(MentorshipRequestDto mentorshipRequestDto) {
+        if (!userRepository.existsById(mentorshipRequestDto.getRequesterId()) ||
+                !userRepository.existsById(mentorshipRequestDto.getReceiverId())) {
+            throw new BusinessException("Один из пользователей не существует.");
```
EntityNotFoundException + указать id, по которым не удалось найти сущность + разделить проверки на существование RequesterId и ReceiverId

DmShevchuk commented on src/main/java/school/faang/user_service/service/MentorshipRequestService.java:
```diff
@@ -0,0 +1,108 @@
+    @Transactional
+    public void rejectRequest(long id, RejectionDto rejection) {
+        MentorshipRequest request = mentorshipRequestRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("Запрос на менторство не найден."));
```
ЕнтитиНотФаундException

DmShevchuk commented on src/main/java/school/faang/user_service/service/MentorshipRequestService.java:
```diff
@@ -0,0 +1,108 @@
+                .orElseThrow(() -> new RuntimeException("Запрос на менторство не найден."));
+
+        if (request.getStatus() == RequestStatus.REJECTED) {
+            throw new IllegalArgumentException("Запрос уже отклонен.");
```
BusinessException

MaxDrattcev committed:
Запрос на менторство: JS2-53972.
- Изменил URL эндпоинтов;
- В контролере поменял возвращаемый результат методов (с ResponseEntity на Dto);
- Добавил исключение EntityNotFoundException;
- Разделил проверки на существование RequesterId и ReceiverId;
- Обновил тесты.
